### DISCUSSION TITLE
Components library utilities

### DISCUSSION THREAD

**Discussion by JMartinCollins** [edmundhung/conform#834] [Q&A]
I've been working with Conform using Shadcn/Radix UI form components. The biggest issue that I have run into with this combo is that the Conform `get<Element>Props` utils don't work with these components. Trying to figure out how to get them to integrate with Conform can be quite a headache. For example, using the Switch component requires understanding how Radix uses a hidden input to include the form value 'on' while the Switch is controlled with a boolean value and a non standard prop `onCheckedChange`. 

Is anyone maintaining utilities like `getRadixSwitchProps` for specific component libraries like these to generate the necessary props easily?

**Comment by updbqn** [comment]
Have you looked at the shadcn examples?

https://conform.guide/examples/shadcn-ui/

Switch example:

https://github.com/edmundhung/conform/blob/b0fb91dbab474c7a150d5f56cb71dbb92888779c/examples/shadcn-ui/src/components/conform/Switch.tsx

I've found that `useControl/useInputControl` makes it possible to integrate most custom components without digging into the internals.
